software/oada_upload/src/oada_upload.ts

```diff
@@ -0,0 +1,387 @@
+// in case it boots before the process filling it but does not write to it
+//
+// One db "owned" by this process. It is linked to the GPS DB using triggers or foreign
+// keys + cascades on update/delete. This is used by this process to keep track of which
+// messages are sent and which are not
+
+// Definition of an interface (similar to an object) used
+// to organize data pulled from the database
+interface DataIndex {
+  [key: string]: {
+    epochs: Array<string>;
+    timetzs: Array<string>;
+    locations: {
+      [key: string]: Omit<Location, Location['_type']>;
+    };
+  };
+}
+
+// Main function
+async function main(): Promise<void> {
+
+  // Pull options from enviroment variables which are defined
+  // in a *.env file or docker-compose.yml. This is the nominal
+  // way to pass options with docker. Do basic type checking
+  // TODO: Further option validation, docker secrets
+  const batchsize = process.env.oada_server_batchsize;
+  assert.numericString(batchsize);
+  const id = process.env.isoblue_id;
+  assert.string(id);
+  const domain = process.env.oada_server_url;
+  assert.string(domain);
+  const token = process.env.oada_server_token;
+  assert.string(token);
+  const db_user = process.env.db_user;
+  assert.string(db_user);
+  const db_host = process.env.db_host;
+  assert.string(db_host);
+  const db_database = process.env.db_database;
+  assert.string(db_database);
+  const db_password = process.env.db_password;
+  assert.string(db_password);  // Passwords with special chars have caused issues in the past
+  const db_port = Number(process.env.db_port);
+  assert.integer(db_port);
+
+
+  // Print out options read in from enviromental variables for manual validation
+  console.log(`Options from enviroment lets:
+    Batch size: ${batchsize}
+    ID: ${id}
+    Domain: ${domain}
+    Token: ${token}
+    DB User: ${db_user}
+    DB Host: ${db_host}
+    DB Database: ${db_database}
+    DB Password ${db_password}
+    DB Port ${db_port}`);
+
+  // Open a connection to the database using pg-node
+  // Previously used a connection string, but this did
+  // not seem to work
+  console.log('Creating databse client');
+  const db = new Client({
+    user: db_user,
+    host: db_host,
+    database: db_database,
+    password: db_password,
+    port: db_port,
+  })
+  console.log('Connecting to database');
+
+  // With the db client setup, initiate a connection to the database
+  try{ 
+    await db.connect();
+  }catch (e){
+    // The databse is hosted on the docker network, so there really shouldn't be any reason we
+    // cannot connect. If we cannot connect this could mean either there is a severe 
+    // misconfiguration (in which we cannot do much) or the db is still booting 
+    // (Happens occasionally when the db booting for the first time). 
+    // TODO: If the DB is still booting, exiting to restart is fine, but should 
+    // investigate if there are ways to detect this wait before connecting
```

Comment:
Was hoping to use a modified `wait-for-postgres.sh` that they gave in those docs, however this container does not has `psql` installed and `psql` typically requires an entire postgres installation from what I can tell. Will look into a `wait-for.sh` instead

Reply:
I see. It is nice to have an application specific health-check ... but I'm not sure its worth needing to ship psql around and expose any cruff that it may be carrying. 

I assume postgres will accept a query after opening the port?

Reply:
I don't know for sure but it would make sense. Worst case the this process exits, restarts, and tries again

Reply:
:+1: